src/tuf.rs

```diff
@@ -395,6 +395,50 @@ impl<D: DataInterchange> Tuf<D> {
         Ok(true)
     }
 
+    /// Walk the base targets delegations and any known nested delegations for the authorized
+    /// signing keys and metadata for the delegation given by `role`.
+    fn find_delegation(&self, role: &MetadataPath) -> Vec<(Vec<&PublicKey>, &Delegation)> {
```

Comment:
Good old tree searching. Unfortunately delegations could potentially have a loop, so can you add in a "I visited this node already" set to avoid infinite loops?

Reply:
Of course, why wouldn't that be supported :)

Fixed by finding just the delegation we are looking for without having to walk the tree.